src/lib.rs

```diff
@@ -17,6 +17,37 @@ pub fn fpm_ftd() -> &'static str {
     include_str!("../fpm.ftd")
 }
 
+pub fn never_expected<'de, D, T>(_deserializer: D) -> std::result::Result<T, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+{
+    unreachable!("must never happen")
+}
+
+#[derive(thiserror::Error, Debug, serde_derive::Deserialize)]
+pub enum Error {
+    #[serde(deserialize_with = "never_expected")]
+    #[error("HttpError: {}", _0)]
+    HttpError(#[from] reqwest::Error),
+
+    #[serde(deserialize_with = "never_expected")]
+    #[error("IoError: {}", _0)]
+    IoError(#[from] std::io::Error),
+
+    #[serde(deserialize_with = "never_expected")]
+    #[error("IoError: {}", _0)]
+    ZipError(#[from] zip::result::ZipError),
+
+    #[error("{doc_id}:{line_number} -> {message}")]
+    ParseError {
+        message: String,
+        doc_id: String,
+        line_number: usize,
+    },
```

Comment:
Would we get this error? Technically we are parsing things, and there can be errors. But largely most errors would come from `ftd`.